src/zenml/integrations/huggingface/services/huggingface_deployment.py

```diff
@@ -170,39 +193,30 @@ def check_status(self) -> Tuple[ServiceState, str]:
             providing additional information about that state (e.g. a
             description of the error, if one is encountered).
         """
-        # TODO: Support all different InferenceEndpointStatus
         try:
-            _ = self.hf_endpoint.status
-        except (InferenceEndpointError, HfHubHTTPError):
-            return (ServiceState.INACTIVE, "")
-
-        if self.hf_endpoint.status == InferenceEndpointStatus.RUNNING:
-            return (
-                ServiceState.ACTIVE,
-                "Hugging Face Inference Endpoint deployment is available",
-            )
-
-        elif self.hf_endpoint.status == InferenceEndpointStatus.SCALED_TO_ZERO:
-            return (
-                ServiceState.ACTIVE,
-                "Hugging Face Inference Endpoint deployment is scaled to zero",
-            )
-
-        elif self.hf_endpoint.status == InferenceEndpointStatus.FAILED:
-            return (
-                ServiceState.ERROR,
-                "Hugging Face Inference Endpoint deployment failed: ",
-            )
+            status = self.hf_endpoint.status
+            if status == InferenceEndpointStatus.RUNNING:
+                return (ServiceState.ACTIVE, "")
+
+            elif status == InferenceEndpointStatus.SCALED_TO_ZERO:
+                return (
+                    ServiceState.SCALED_TO_ZERO,
+                    "Hugging Face Inference Endpoint is scaled to zero, but still running. It will be started on demand.",
+                )
 
-        elif self.hf_endpoint.status == InferenceEndpointStatus.PENDING:
+            elif status == InferenceEndpointStatus.FAILED:
+                return (
+                    ServiceState.ERROR,
+                    "Hugging Face Inference Endpoint deployment failed: ",
+                )
+            elif status == InferenceEndpointStatus.PENDING:
+                return (ServiceState.PENDING_STARTUP, "")
+            return (ServiceState.PENDING_STARTUP, "")
+        except (InferenceEndpointError, HfHubHTTPError):
             return (
-                ServiceState.PENDING_STARTUP,
-                "Hugging Face Inference Endpoint deployment is being created: ",
+                ServiceState.INACTIVE,
+                "Hugging Face Inference Endpoint deployment is inactive or not found",
```

Comment:
One of these ends in `: ` and the other has no space or colon. What is the correct formatting needed for these error messages? (I'm not sure where they get returned up into... are they printed to the screen somewhere?